### PR TITLE
Add user booking local database

### PR DESCRIPTION
**Pull request by Yuhuu**
Add user booking local database

### CODE REVIEW

File: server/Context/OfficeDbContext.cs [hunk 1]
```diff
@@ -1,133 +1,25 @@
...
-        public DbSet<User> Users { get; set; }
-        public DbSet<Seat> Seats { get; set; }
-        public DbSet<Room> Rooms { get; set; }
-        public DbSet<Booking> Bookings { get; set; }
-
-        public OfficeDbContext(DbContextOptions options) : base(options)
-        {
-        }
-
-        protected override void OnModelCreating(ModelBuilder modelBuilder)
```
Comment (sakerchr) [inline server/Context/OfficeDbContext.cs:17]: This part is very important and should not be removed

File: server/Properties/ServiceDependencies/AppModelv2-WebApp-OpenIDConnect-DotNet20231017 - Web Deploy/profile.arm.json
Comment (sakerchr) [inline server/Properties/ServiceDependencies/AppModelv2-WebApp-OpenIDConnect-DotNet20231017 - Web Deploy/profile.arm.json:1]: Why are we adding arm templates?

File: server/Program.cs [hunk 2]
```diff
@@ -67,63 +63,40 @@
     options.DefaultPolicy = defaultPolicy;
     options.FallbackPolicy = defaultPolicy;
 });
-builder.Services.AddControllersWithViews();
 
-builder.Services.AddCors(options =>
-{
+// TODO
```
Comment (sakerchr) [inline server/Program.cs:67]: If this is for having a local database, do a check on whether or not the environment is development, and set DbContext based on this

File: server/Program.cs [hunk 3]
```diff
@@ -67,63 +63,40 @@
...
 
+app.MapSeatEndpoints();
+app.MapUserEndpoints();
+app.MapBookingEndpoints();
+app.UseRouting();
+app.UseAuthorization();
+app.UseEndpoints(e => e.MapDefaultControllerRoute());
 app.MapFallbackToFile("index.html");
 
-// Temporary fix to apply migrations on startup
```
Comment (sakerchr) [inline server/Program.cs:119]: Dont remove this, we need it.

File: server/Program.cs [hunk 4]
```diff
@@ -82,48 +88,57 @@
...
-    name: "default",
-    pattern: "{controller}/{action=Index}/{id?}");
+//app.UseCors(p => p.WithOrigins("http://localhost:3000").AllowAnyHeader().AllowAnyMethod());
 
+app.MapSeatEndpoints();
+app.MapUserEndpoints();
+app.MapBookingEndpoints();
+app.UseRouting();
+app.UseAuthorization();
+app.UseEndpoints(e => e.MapDefaultControllerRoute());
```
Comment (mostafasaziz) [inline server/Program.cs:128]: Her får jeg en warning som klager på følgende: 
https://learn.microsoft.com/en-us/aspnet/core/diagnostics/asp0014?view=aspnetcore-8.0

File: server/Program.cs [hunk 5]
```diff
@@ -82,48 +88,57 @@
...
 // Temporary fix to apply migrations on startup
 // since we dont apply them in our pipeline
 // (inb4 this is a permanent fix)
-using (var scope = app.Services.CreateScope())
+if (builder.Environment.IsProduction())
+{
+    using (var scope = app.Services.CreateScope())
 {
-    var dbContext = scope.ServiceProvider.GetRequiredService<OfficeDbContext>();
+    var dbContext = scope.ServiceProvider.GetRequiredService<server.Context.OfficeDbContext>();
```
Comment (mostafasaziz) [inline server/Program.cs:138]: Trenger vi å spesifisere med server.Context her? Funker det å bare bruke OfficeDbContext?

File: server/Program.cs (hunk 2 above)
Comment (Yuhuu) [reply server/Program.cs]: added environment check

File: server/Properties/ServiceDependencies/AppModelv2-WebApp-OpenIDConnect-DotNet20231017 - Web Deploy/profile.arm.json
Comment (Yuhuu) [reply server/Properties/ServiceDependencies/AppModelv2-WebApp-OpenIDConnect-DotNet20231017 - Web Deploy/profile.arm.json:1]: removed

File: server/Context/OfficeDbContext.cs (hunk 1 above)
Comment (Yuhuu) [reply server/Context/OfficeDbContext.cs:17]: take back the changed on dbcontext to prod

File: server/Program.cs (hunk 3 above)
Comment (Yuhuu) [reply server/Program.cs:123]: take back this

File: server/Program.cs (hunk 5 above)
Comment (Yuhuu) [reply server/Program.cs:171]: Det funger bra uten server.Context, fjerne server.Context her nå

File: omega-office-booking.sln [hunk 6]
```diff
@@ -13,8 +13,8 @@ Global
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
 		{B9775CEF-C3D9-4991-BE1C-892E8252EFBD}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{B9775CEF-C3D9-4991-BE1C-892E8252EFBD}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{B9775CEF-C3D9-4991-BE1C-892E8252EFBD}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{B9775CEF-C3D9-4991-BE1C-892E8252EFBD}.Release|Any CPU.Build.0 = Release|Any CPU
+		{B9775CEF-C3D9-4991-BE1C-892E8252EFBD}.Release|Any CPU.ActiveCfg = Debug|Any CPU
```
Comment (sakerchr) [inline omega-office-booking.sln:16]: this seems like something that was forgotten about before pushing?

File: server/Context/OfficeDbContextLokal.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
+using Microsoft.EntityFrameworkCore;
+using server.Data;
+using server.Models.Domain;
+
+namespace server.Context;
+
+    public class OfficeDbContextLokal : DbContext
```
Comment (sakerchr) [inline server/Context/OfficeDbContextLokal.cs:7]: Lokal => Local ?

File: server/Data/BookingRepository.cs [hunk 8]
```diff
@@ -0,0 +1,50 @@
+using Microsoft.EntityFrameworkCore;
+using server.Context;
+using server.Data;
+using server.Models.Domain;
+using System.Globalization;
+using System.Linq;
+
+public interface IBookingRepository
```
Comment (sakerchr) [inline server/Data/BookingRepository.cs:8]: Typically you move interfaces to their own file

File: server/Data/RoomRepository.cs [hunk 9]
```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.EntityFrameworkCore;
+using server.Context;
+using server.Models.Domain;
+
+namespace server.Data
+{
+    public interface IRoomRepository
```
Comment (sakerchr) [inline server/Data/RoomRepository.cs:7]: Typically you move interfaces to their own file

File: server/Data/SeatRepository.cs [hunk 10]
```diff
@@ -0,0 +1,83 @@
+﻿using Microsoft.EntityFrameworkCore;
+using server.Context;
+using server.Models.Domain;
+
+namespace server.Data
+{
+    public interface ISeatRepository
```
Comment (sakerchr) [inline server/Data/SeatRepository.cs:7]: Typically you move interfaces to their own file

File: server/Data/UserRepository.cs [hunk 11]
```diff
@@ -0,0 +1,81 @@
+﻿using Microsoft.EntityFrameworkCore;
+using server.Context;
+using server.Models.Domain;
+
+namespace server.Data
+{
+    public interface IUserRepository
```
Comment (sakerchr) [inline server/Data/UserRepository.cs:7]: Typically you move interfaces to their own file

File: server/Context/OfficeDbContextLokal.cs (hunk 7 above)
Comment (Yuhuu) [reply server/Context/OfficeDbContextLokal.cs:7]: typo, for fort gjord feil med norsk/english

**Pull request closed without merging**